### PR TITLE
Restyle Chromium enable notifications dialog

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/25593

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

image

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
+package org.chromium.chrome.browser.notifications.permissions;
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java:1]: Please add our license header here

File: build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+public class BraveNotificationPermissionRationaleDialogControllerClassAdapter extends BraveClassVisitor {
+    static String sContoller =
+            "org/chromium/chrome/browser/notifications/permissions/NotificationPermissionRationaleDialogController";
+    static String sBraveContoller =
+            "org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController";
+
+    public BraveNotificationPermissionRationaleDialogControllerClassAdapter(ClassVisitor visitor) {
+        super(visitor);
+
+        redirectConstructor(sContoller, sBraveContoller);
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java:19]: Please add check to `testConstructorsExistAndMatch` in `BytecodeTest.java`

Comment (samartnik) [reply build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java:19]: And btw, for test to work in release we will need to add proguard rule to `brave/android/java/apk_for_test.flags` as well.

File: android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        Button btnNotNow = dialogView.findViewById(R.id.notification_not_now_button);
+        btnNotNow.setOnClickListener((v) -> {
+            mModalDialogManager.dismissDialog(
+                    dialogModel, DialogDismissalCause.NEGATIVE_BUTTON_CLICKED);
+        });
+
+        mModalDialogManager.showDialog(dialogModel, ModalDialogType.APP);
+    }
+
+    private Callback<Integer> wrapDialogDismissalCallback(Callback<Boolean> rationaleCallback) {
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java:82]: This seems to be 1 to 1 copy of the parent's implementation. If so, we should use parent's call directly. See `destroyPropertiesForStream` as an example.

File: android/brave_java_resources.gni [hunk 4]
```diff
@@ -710,6 +712,9 @@ brave_java_resources = [
   "java/res/font/poppins_light.ttf",
   "java/res/font/poppins_medium.ttf",
   "java/res/font/poppins_regular.ttf",
+  "java/res/font/roboto_bold.ttf",
```
Comment (samartnik) [inline android/brave_java_resources.gni:715]: Are these fonts a requirement from the design team?

File: build/android/bytecode/java/org/brave/bytecode/BraveClassAdapter.java [hunk 5]
```diff
@@ -81,6 +81,7 @@ public static ClassVisitor createAdapter(ClassVisitor chain) {
         chain = new BraveVariationsSeedFetcherClassAdapter(chain);
         chain = new BraveWebsiteClassAdapter(chain);
         chain = new BraveWebsitePermissionsFetcherClassAdapter(chain);
+        chain = new BraveNotificationPermissionRationaleDialogControllerClassAdapter(chain);
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveClassAdapter.java:84]: Please sort it alphabetically.

**Review by samartnik — CHANGES_REQUESTED** [review]

File: android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java (hunk 3 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java:82]: even 1 to 1 copy but it's private to that class.
and public method showRationaleUi it's changed all , so no need to call super

File: android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java (hunk 1 above)
Comment (sujitacharya2005) [reply android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java:6]: done

File: build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java (hunk 2 above)
Comment (sujitacharya2005) [reply build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java:20]: updated apk_for_test.flags and ByteCodeTest.java

File: build/android/bytecode/java/org/brave/bytecode/BraveClassAdapter.java (hunk 5 above)
Comment (sujitacharya2005) [reply build/android/bytecode/java/org/brave/bytecode/BraveClassAdapter.java]: done

File: android/brave_java_resources.gni (hunk 4 above)
Comment (samartnik) [reply android/brave_java_resources.gni:715]: @brave/design-team is this a requirement? Wouldn't it be inconsistent to use this font in just one dialog and different font in other UI elements?

File: android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java (hunk 3 above)
Comment (samartnik) [reply android/java/org/chromium/chrome/browser/notifications/permissions/BraveNotificationPermissionRationaleDialogController.java:84]: We can make it public in bytecode and I don't really see any advantage of having a copy of it. Are we planing to extend it?

File: android/brave_java_resources.gni (hunk 4 above)
Comment (Sam-Sibley) [reply android/brave_java_resources.gni:715]: @samartnik This should match others. @sujitacharya2005 , can you confirm it is correct fonts.

Comment (sujitacharya2005) [reply android/brave_java_resources.gni:715]: Thanks @samartnik pointing out
In figma it's mentioned Roboto font.
I discussed with @Sam-Sibley regarding this
As per @Sam-Sibley it's used across brave and rewards now it's used poppins font.
We can use these options which is same as Roboto, so no need add these newly added fonts.
android:fontFamily="sans-serif"
android:fontFamily="sans-serif-light"
android:fontFamily="sans-serif-condensed"
android:fontFamily="sans-serif-thin"
android:fontFamily="sans-serif-medium"

So I will remove these newly added fonts.

File: build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        deleteMethod(sBraveContoller, "wrapDialogDismissalCallback");
+        makePublicMethod(sContoller, "wrapDialogDismissalCallback");
```
Comment (samartnik) [inline build/android/bytecode/java/org/brave/bytecode/BraveNotificationPermissionRationaleDialogControllerClassAdapter.java:22]: Please add a check to `testMethodsExist` and appropriate proguard rule.

**Review by samartnik — APPROVED** [review]
lgtm, just 1 more test required

**Review by tapanmodh — APPROVED** [review]
LGTM

**Review by deeppandya — APPROVED** [review]
LGTM

**Comment by deeppandya** [comment]
@sujitacharya2005 we may need to uplift these changes. Let me know if I can help with uplift

**Pull request merged**